pkg/backup/backup.go

```diff
@@ -407,6 +407,12 @@ func (kb *kubernetesBackupper) BackupWithResolvers(log logrus.FieldLogger,
 	// no more progress updates will be sent on the 'update' channel
 	quit <- struct{}{}
 
+	// For some use cases, e.g. use label to filter resources in backup,
+	// namespace-scoped resources are included, but the namespace contains
+	// the resources are not. Add the function to include the namespaces
+	// if needed.
+	kb.backupAdditionalNamespaces(items, itemBackupper, log)
```

Comment:
Does the logic of this function conflict with the one in https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/item_collector.go#L282? 
How about removing the label-matching logic for namespaces directly?

Reply:
In most cases, removing this code should have no change in behavior.
I only think one scenario needs consideration.
If the user labels the namespace resource and no resources in that namespace are selected in the backup, then the namespace will not be included in the backup.

Reply:
After reading the https://github.com/vmware-tanzu/velero/blob/main/pkg/backup/item_collector.go cluster scope handling code, I agree with Wenkai.
Namespace resources are already handled correctly in the cluster-scoped resource code.